tuna/connection.py

```diff
@@ -174,14 +257,17 @@ def exec_command_unparsed(self, cmd, timeout=SSH_TIMEOUT, abort=None):
       if abort is not None and chk_abort_file(self.id, self.logger):
         self.logger.warning('Machine %s aborted command execution: %s', self.id,
                             cmd)
-        return None, None, None
+        return i_var, o_var, e_var
 
     self.logger.error('cmd_exec retries exhausted, giving up')
-    return None, None, None
+    return i_var, o_var, e_var
```

Comment:
260, 263 I don't see a path where these values are initialized before these return statements. How is this not a pylint error?

Reply:
pylint checks are passing.

Reply:
These values need to be explicitly initialized before they are used as returns. (set to None)

Reply:
If you can verify valid behavior from the users of this function you could also set it as an empty IO / Channel object.

Reply:
In code base, we just log the error messages. and in most of the case we look for mid paramenter that is output file type. some cases , we are checking for exit_recev_status/0ready(), which we are calling already. no where we it is set to empty object type. Eiter truncate/seek(0) can be used.